### ISSUE TITLE
Autocompletion slow

### ISSUE THREAD

**Issue by LuRsT** [fish-shell/fish-shell#6592]
Hello Fish maintainers!

## Context

Here, here are my specs:

```
$ fish --version
fish, version 3.0.2
$ echo $TERM
tmux-256color
```

Using Arch linux in a kitty term, and I've got an issue where at times, fish is slow at tab completing, I can't reproduce the exact environment, but I know that if I open a new tab, this happens, wether inside tmux or not. This situation happens in a terminal where I've spent some quality time.

## The issue

I get the completion from fish correctly, but when I press "Arrow right" or "Tab", it takes about 3-5seconds to finish writing the missing letters, it's also blocking everything in that terminal (although the cursor still blinks), so I can't even Contrl+C out of that operation.

## Question

How can I best figure out what is the problem?

**Comment by zanchey** [comment]
Running under `strace` or using `fish --profile somefile.profile` might be the best place to start.

If you've not used `strace` before, a command like `strace -tt -T -o myfishtrace.txt fish` will give you a useful output file in `myfishtrace.txt`, which you can then cut down to the lines around the time of the hang.

**Comment by LuRsT** [comment]
Thank you very much for the insightful answer @zanchey, that's exactly what I was looking for, when I manage to replicate it again, I'll update this issue with more information.

**Comment by LuRsT** [comment]
I did it, and managed to figure out how to replicate it with a single command, this may help me understand what the problem is. The other problem I have now is that the profile file is 10K lines long (which may be a signal that something is going wrong).

This is how I get that problem:
- `cd` into a directory with a `.git` folder (as far as I can tell, that's what the folders have in common)
- Type `vim REA` (trying to edit the README.md file) and when I tab, it stops for a few seconds, and it's slow even afterwards when I try to Ctrl+C.

Anything you can recognize as being a problem while I peruse the strace file?

**Comment by faho** [comment]
@LuRsT Do you have vim aliased to anything?

I'm assuming you do, and it's nvim, and then completing `vim` tries to load the `nvim` completions, which load the `vim` completions.

In other words this looks like a duplicate of #5949, which is fixed in 3.1.0, which was just released.

**Comment by LuRsT** [comment]
> @LuRsT Do you have vim aliased to anything?

There's no shortage for aliases in my config :D, I did recently add an alias for `alias vim=nvim`. I read the thread you linked and YUP! That's it, I removed the aliases and it's all good now, thank you so much! And now I know more about `strace` and how `fish` handles aliases (which I think I need to read a bit more...).

Thank you very much!
